### ISSUE TITLE
error: Get http://unix.socket/1.0: dial unix /var/lib/lxd/unix.socket: connection refused

### ISSUE THREAD

**Issue by techtonik** [lxc/lxd#275]
`lxc list`, `lxc finger` and other commands give this error.

    error: Get http://unix.socket/1.0: dial unix /var/lib/lxd/unix.socket: connection refused

Installed from PPA on Ubuntu 14.04 x32, reports version 0.1

**Comment by stgraber** [comment]
Your user must be in the lxd group for it to have access to lxd (similar to libvirtd for libvirt).

On package installation, all members of the admin and sudoers group are automatically added to the lxd group. However groups are only effective after a new session is open.

So you can either close your user session and open a new one or run "newgrp lxd" in your shell prior to running any lxc command.

**Comment by techtonik** [comment]
@stgraber, the result is the same with sudo.

**Comment by stgraber** [comment]
Oh, that's a bit weirder then

**Comment by techtonik** [comment]
Tried with `newgrp lxd`. Same problem:
```
$ newgrp lxd
$ sudo lxc list
error: Get http://unix.socket/1.0: dial unix /var/lib/lxd/unix.socket: connection refused
$ groups
lxd adm cdrom sudo dip plugdev lpadmin sambashare techtonik
$ lxc list
error: Get http://unix.socket/1.0: dial unix /var/lib/lxd/unix.socket: connection refused
```

**Comment by stgraber** [comment]
What do you get from "sudo status lxd" ?

**Comment by techtonik** [comment]
```
$ sudo status lxd
lxd stop/waiting
```

**Comment by techtonik** [comment]
```
$ dmesg | grep lxd
[ 2784.103166] init: lxd main process (1145) terminated with status 1
[ 2784.103191] init: lxd main process ended, respawning
[ 2784.143774] init: lxd main process (1150) terminated with status 1
...
[ 2784.430497] init: lxd main process (1199) terminated with status 1
[ 2784.430521] init: lxd main process ended, respawning
[ 2784.469444] init: lxd main process (1205) terminated with status 1
[ 2784.469467] init: lxd respawning too fast, stopped
```

**Comment by stgraber** [comment]
Ok, so looks like lxd is crashing for you.

What do you have in /var/log/upstart/lxd.log ?

**Comment by techtonik** [comment]
```
$ sudo less /var/log/upstart/lxd.log
/var/log/upstart/lxd.log: No such file or directory
```

**Comment by techtonik** [comment]
```
$ locate lxd.log
/var/log/upstart/lxd.log.1.gz
```

**Comment by stgraber** [comment]
Hmm, ok.

Then what do you get if you manually run:

    sudo lxd --group lxd

**Comment by techtonik** [comment]
```
$ sudo lxd --group lxd
error: cannot listen on unix socket: listen unix /var/lib/lxd/unix.socket: bind: address already in use
```

**Comment by stgraber** [comment]
Anything interesting in "ps aux | grep lxd" ?

**Comment by techtonik** [comment]
Nope.
```
$ sudo ps aux | grep lxd
root     24888  0.0  0.0   6148   940 pts/0    S    23:07   0:00 newgrp lxd
techton+ 25060  0.0  0.0   6156   812 pts/0    S+   23:15   0:00 grep --color=auto lxd
```

**Comment by stgraber** [comment]
sudo netstat -lnp | grep lxd

**Comment by techtonik** [comment]
Empty.
```
$ sudo netstat -lnp | grep lxd
$
```

**Comment by stgraber** [comment]
Hmm, so nothing's using the socket but it's already in use? That doesn't really make any sense.

**Comment by techtonik** [comment]
`lxd.log.1.gz` contains same lines with error: cannot listen on unix socket

**Comment by techtonik** [comment]
```
$ ls -la /var/lib/lxd
total 20
drwxr-xr-x  3 root root 4096 Feb 19 11:14 .
drwxr-xr-x 90 root root 4096 Feb 26 10:03 ..
drwxr-xr-x  2 root root 4096 Feb 18 22:16 lxc
-rw-r--r--  1 root root 1805 Feb 18 22:17 server.crt
-rw-------  1 root root 3247 Feb 18 22:17 server.key
srwxrwxr-x  1 root lxd     0 Feb 19 11:14 unix.socket
```

**Comment by stgraber** [comment]
dmesg | tail -n 10

**Comment by techtonik** [comment]
```
[198530.000068] firewire_core 0000:04:03.0: rediscovered device fw0
[198539.488023] ata1: softreset failed (device not ready)
[198546.252027] ata1: SATA link up 3.0 Gbps (SStatus 123 SControl 300)
[198546.264240] ata1.00: configured for UDMA/133
[198546.280134] sd 0:0:0:0: [sda] Starting disk
[198546.283852] PM: resume of devices complete after 16847.475 msecs
[198546.284269] PM: Finishing wakeup.
[198546.284272] Restarting tasks ... done.
[198546.996807] atl1 0000:02:00.0: irq 45 for MSI/MSI-X
[198546.996895] atl1 0000:02:00.0: eth1 link is up 100 Mbps full duplex
```

**Comment by stgraber** [comment]
hmm, maybe try removing /var/lib/lxd/unix.socket and run lxd --group lxd manually again?

**Comment by stgraber** [comment]
I just installed lxd i386 from the PPA here on a 32bit system and it works as expected so I'm not too sure what's different with your system.

**Comment by techtonik** [comment]
```
$ rm /var/lib/lxd/unix.socket 
rm: cannot remove ‘/var/lib/lxd/unix.socket’: Permission denied
$ sudo rm /var/lib/lxd/unix.socket 
$ lxd --group lxd
error: cannot listen on unix socket: listen unix /var/lib/lxd/unix.socket: bind: permission denied
$ sudo lxd --group lxd
```
Now it waits.

**Comment by stgraber** [comment]
ok, that's more like it. Can you talk to it with lxc?

**Comment by techtonik** [comment]
Looks like there is no error now. How can I create a container without scripts from #276?
```
$ lxc finger
$ lxc list
$
```

**Comment by stgraber** [comment]
Hmm, actually with 0.1 on a 32bit system, I'm reasonably sure you can't :)

That's because we didn't have images back then and so a 64bit Ubuntu container is hardcoded.

You can still try with: lxc launch ubuntu test

But I'm pretty sure it'll get created fine and will then fail to start because you're running on 32bit.

**Comment by stgraber** [comment]
oh and now that lxd seems to be behaving a bit better (not sure what was going on before). You can probably start it properly by ctrl+c-ing the one you manually started and by running "sudo start lxd" to start it properly again.

**Comment by techtonik** [comment]
=))

Restarted with `sudo start lxd`. Server seems working ok - when I turn `--debug -v` I see there is communication with it like `2015/02/26 23:30:35 pong received`.

**Comment by techtonik** [comment]
```
$ lxc launch ubuntu test
Creating container...error: shared's user has no subuids
```
Is that the result of aforementioned 32bit/64bit mismatch error?

**Comment by stgraber** [comment]
Ah no, that one isn't. That one is because you don't have a uid and gid alllocation for root in /etc/subuid and /etc/subgid.

The 0.2 package will do that automatically for you, but until then, the easiest is to edit those two files by hand and copy your own user's line, replacing your username by root.

**Comment by techtonik** [comment]
These uid/gid allocations sound complicated. I also see that `amd64` container is requested, so it won't help anyway. Maybe it is better to ask when 0.2 is planned to be released to PPA? Maybe there are some daily builds in `lxd-daily` PPA? =)
```
$ lxc launch ubuntu test -v --debug
Creating container...2015/02/26 23:36:03 fingering the daemon
2015/02/26 23:36:03 raw response: {"type":"sync","status":"Success","status_code":200,"metadata":{"api_compat":1,"auth":"trusted","config":[{"key":"trust-password","value":false}],"environment":{"backing_fs":"ext4","driver":"lxc","kernel_version":"3.13.0-45-generic","lxc_version":"1.1.0"}}}
2015/02/26 23:36:03 pong received
2015/02/26 23:36:03 posting {"name":"test","source":{"name":"lxc-images/ubuntu/trusty/amd64","type":"remote","url":"https+lxc-images://images.linuxcontainers.org"}}
 to http://unix.socket/1.0/containers
2015/02/26 23:36:03 raw response: {"error":"shared's user has no subuids","error_code":400,"type":"error"}

error: shared's user has no subuids
```

**Comment by techtonik** [comment]
The subuid and subgid files are empty. What should be there?
```
$ cat /etc/subuid
# empty default subuid/subgid file
$ cat /etc/subgid
# empty default subuid/subgid file
```

**Comment by stgraber** [comment]
Hi,

Sorry for the delay.

We've now released 0.2 and it's packaged in our PPA, including support for x86 32bit and with a postinst script which will setup the subuid and subgid maps for you. Can you try that and see if that works for you?

**Comment by stgraber** [comment]
The socket re-use problem mentioned in this issue was resolved by Tycho this morning.

**Comment by techtonik** [comment]
Let me test this..

**Comment by techtonik** [comment]
Got a crash while updating to `lxd` 0.2. I guess I need to reinstall from scratch.

**Comment by techtonik** [comment]
`apt-get purge lxd lxc; apt-get autoremove`

**Comment by techtonik** [comment]
Installed. After reboot the original message is gone. At least there is no error now.
```
$ lxc version
0.2
$ lxc list
$ lxc finger
```

Thanks for helping with that. I will try to create containers now and open new reports if new issues appear.

**Comment by techtonik** [comment]
Fixed in lxd 0.2.

**Comment by SimplySeth** [comment]
Ubuntu 16.04.2 LTS

lxd                                 2.0.9

lxd-client                      2.0.9

I get this when doing `sudo lxd init`

`error: Unable to talk to LXD: Get http://unix.socket/1.0: dial unix /var/lib/lxd/unix.socket: connect: connection refused`


 /var/lib/lxd looks like this 

```drwx--x--x 2 root lxd    6 Mar 15 15:43 containers
drwx--x--x 2 root lxd    6 Mar 15 15:43 devices
drwxr-xr-x 2 root lxd    6 Mar 15 15:44 devlxd
drwx------ 2 root lxd    6 Mar 15 15:43 images
-rw-r--r-- 1 root lxd  41K Mar 15 15:44 lxd.db
drwx------ 2 root lxd    6 Mar 15 15:43 security
-rw-r--r-- 1 root lxd 1.9K Mar 15 15:43 server.crt
-rw------- 1 root lxd 3.2K Mar 15 15:43 server.key
drwx--x--x 2 root lxd    6 Mar 15 15:43 shmounts
drwx------ 2 root lxd    6 Mar 15 15:43 snapshots
srw-rw---- 1 root lxd    0 Mar 15 16:01 unix.socket
-rw------- 1 root lxd 2.0G Mar 15 16:01 zfs.img
```
